src/components/AutoUpdateTime.js

```diff
@@ -22,85 +28,62 @@ const propTypes = {
      */
-    getCurrentUserLocalTime() {
-        return DateUtils.getLocalMomentFromDatetime(
-            this.props.preferredLocale,
+    const getCurrentUserLocalTime = useCallback(() => (
+        DateUtils.getLocalMomentFromDatetime(
+            props.preferredLocale,
             null,
-            this.props.timezone.selected,
-        );
-    }
+            props.timezone.selected,
+        )
+    ), [props.preferredLocale, props.timezone.selected]);
 
-    /**
-     * @returns {string} Returns the timezone name in string, e.g.: GMT +07
-     */
-    getTimezoneName() {
+    const [currentUserLocalTime, setCurrentUserLocalTime] = useState(getCurrentUserLocalTime);
+    const minuteRef = useRef(new Date().getMinutes());
+    const timezoneName = useMemo(() => {
         // With non-GMT timezone, moment.zoneAbbr() will return the name of that timezone, so we can use it directly.
-        if (Number.isNaN(Number(this.state.currentUserLocalTime.zoneAbbr()))) {
-            return this.state.currentUserLocalTime.zoneAbbr();
+        if (Number.isNaN(Number(currentUserLocalTime.zoneAbbr()))) {
+            return currentUserLocalTime.zoneAbbr();
         }
 
         // With GMT timezone, moment.zoneAbbr() will return a number, so we need to display it as GMT {abbreviations} format, e.g.: GMT +07
-        return `GMT ${this.state.currentUserLocalTime.zoneAbbr()}`;
-    }
+        return `GMT ${currentUserLocalTime.zoneAbbr()}`;
+    }, [currentUserLocalTime]);
 
-    /**
-     * Update the user's local time at the top of every minute
-     */
-    updateCurrentTime() {
-        if (this.timer) {
-            clearTimeout(this.timer);
-            this.timer = null;
-        }
-        const millisecondsUntilNextMinute = (60 - this.state.currentUserLocalTime.seconds()) * 1000;
-        this.timer = setTimeout(() => {
-            this.setState({
-                currentUserLocalTime: this.getCurrentUserLocalTime(),
-            });
-        }, millisecondsUntilNextMinute);
-    }
+    useEffect(() => {
+        // If the user leaves this page open, we want to make sure the displayed time is updated every minute when the clock changes
+        // To do this we create an interval to check if the minute has changed every second and update the displayed time if it has
+        const interval = setInterval(() => {
+            const currentMinute = new Date().getMinutes();
+            if (currentMinute !== minuteRef.current) {
+                setCurrentUserLocalTime(getCurrentUserLocalTime());
+                minuteRef.current = currentMinute;
+            }
+        }, 1000);
+        return () => clearInterval(interval);
+    }, [getCurrentUserLocalTime]);
+
+    useEffect(() => {
+        // If the preferredLocale or timezone changes, we want to update the displayed time immediately
+        setCurrentUserLocalTime(getCurrentUserLocalTime());
+    }, [getCurrentUserLocalTime, props.preferredLocale, props.timezone.selected]);
```

Comment:
Yes, I noticed that too. But the linter doesn't like it when I remove it 😅. Is there any downside to having that redundant dependency? 
<img width="1075" alt="Screenshot 2023-04-11 at 7 42 25 AM" src="https://user-images.githubusercontent.com/1371996/231152607-27b56c7d-480a-4619-a913-c78957fe4702.png">

Reply:
@puneetlath you are updating the wrong line :) I'm referring to the next `useEffect`:

```javascript
useEffect(() => {
  // If the preferredLocale or timezone changes, we want to update the displayed time immediately
  setCurrentUserLocalTime(getCurrentUserLocalTime());
}, [getCurrentUserLocalTime, props.preferredLocale, props.timezone.selected]);
```

The `useEffect` that sets up the interval is correct.

Reply:
I mean, you should remove `props.preferredLocale` and `props.timezone.selected` here, because it is already used as dependency in the `getCurrentUserLocalTime`. So this way: 

```javascript
const getCurrentUserLocalTime = useCallback(() => (
  DateUtils.getLocalMomentFromDatetime(
    props.preferredLocale,
    null,
    props.timezone.selected,
  )
), [props.preferredLocale, props.timezone.selected]);

...

useEffect(() => {
  // If the preferredLocale or timezone changes, we want to update the displayed time immediately
  setCurrentUserLocalTime(getCurrentUserLocalTime());
}, [getCurrentUserLocalTime]);
```

Reply:
Ah got it! That makes sense. Updated.

Reply:
Actually... after the above change there is no sense to have two separate `useEffect's` cause they both have the same dependency array... so we can merge them into single one:

```javascript
useEffect(() => {
  // If the preferredLocale or timezone changes, we want to update the displayed time immediately
  setCurrentUserLocalTime(getCurrentUserLocalTime());

  // If the user leaves this page open, we want to make sure the displayed time is updated every minute when the clock changes
  // To do this we create an interval to check if the minute has changed every second and update the displayed time if it has
  const interval = setInterval(() => {
    const currentMinute = new Date().getMinutes();
    if (currentMinute !== minuteRef.current) {
      setCurrentUserLocalTime(getCurrentUserLocalTime());
      minuteRef.current = currentMinute;
    }
  }, 1000);
  return () => clearInterval(interval);
}, [getCurrentUserLocalTime]);
```

Reply:
Oh yeah, great point! Thanks a lot. Updated.